File: polyglot/overwrite_from_context/py/src/runtime_template_resolver/options.py
from enum import Enum
from dataclasses import dataclass
from typing import Optional
# Avoid circular import if possible. Logger is used for type hint.
# We can use string forward reference or TYPE_CHECKING
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .logger import Logger

class ComputeScope(Enum):
    STARTUP = "STARTUP"
    REQUEST = "REQUEST"

class MissingStrategy(Enum):
    ERROR = "ERROR"
    DEFAULT = "DEFAULT"
    IGNORE = "IGNORE"

@dataclass
class ResolverOptions:
    max_depth: int = 10
    missing_strategy: MissingStrategy = MissingStrategy.ERROR
    logger: Optional['Logger'] = None

File: polyglot/overwrite_from_context/mjs/src/options.ts
export enum ComputeScope {
    STARTUP = 'STARTUP',
    REQUEST = 'REQUEST'
}

export enum MissingStrategy {
    ERROR = 'ERROR',
    DEFAULT = 'DEFAULT',
    IGNORE = 'IGNORE'
}

export interface ResolverOptions {
    logger?: any;
    maxDepth?: number;
    missingStrategy?: MissingStrategy;
}
